sam/models/base_model.py

```diff
@@ -117,79 +118,32 @@ def validate_data(self, X: pd.DataFrame) -> None:
         Validates the data and raises an exception if:
         - There is no time columns
         - The data is not monospaced
-        - There is not enought data
 
         Parameters
         ----------
         x: pd.DataFrame
             The dataframe to validate
         """
         if self.timecol is None:
-            warnings.warn(
-                (
-                    "No timecolumn given. Make sure the data is"
-                    "monospaced when given to this model!"
-                ),
-                UserWarning,
-            )
+            if isinstance(X.index, pd.DatetimeIndex):
+                monospaced = X.index.to_series().diff().dropna().unique().size == 1
+            else:
+                warnings.warn(
+                    (
+                        "No timecolumn given. Make sure the data is"
+                        "monospaced when given to this model!"
+                    ),
+                    UserWarning,
+                )
+                monospaced = True
```

Comment:
So we implicitly assume the data is monospaced? Maybe we should just require this DatetimeIndex?